src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -67,7 +67,7 @@ internal sealed class HttpConnectionPool : IDisposable
         private readonly int _maxConnections;
 
         private bool _http2Enabled;
-        private Http2Connection? _http2Connection;
+        private IReadOnlyList<Http2Connection>? _http2Connections;
```

Comment:
If I get the change right, every time we add a new connection we create a new list +1 size larger, move the existing connections there and swap the private field.
It's maybe worth a few comments in the code.

Reply:
That's correct, but resizing is done in the [dedicated method](https://github.com/dotnet/runtime/blob/d97e37ef01eee58e6d3eb08342ee626d2abf8473/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs#L659) which seems straightforward.

Reply:
However, if it's still not clear, I can add comments directly to this declaration.

Reply:
There is overhead involved when accessing an array that is cast as `IReadOnlyList<T>`. Since this is a private field I would leave it as an array, and add a comment about how to modify it.

Reply:
Agree. Fixed.